src/fides/api/service/connectors/saas_connector.py

```diff
@@ -275,29 +282,75 @@ def retrieve_data(
                     self.secrets,
                 )
 
-            prepared_requests: List[SaaSRequestParams] = query_config.generate_requests(
-                input_data, policy, read_request
-            )
+            # if a path is provided, it means we want to generate HTTP requests from the config
+            if read_request.path:
+                prepared_requests: List[Tuple[SaaSRequestParams, Dict[str, Any]]] = (
+                    query_config.generate_requests(input_data, policy, read_request)
+                )
 
-            # Iterates through initial list of prepared requests and through subsequent
-            # requests generated by pagination. The results are added to the output
-            # list of rows after each request.
-            for next_request in prepared_requests:
-                while next_request:
-                    processed_rows, next_request = self.execute_prepared_request(  # type: ignore
-                        next_request,
-                        privacy_request.get_cached_identity_data(),
-                        read_request,
+                # Iterates through initial list of prepared requests and through subsequent
+                # requests generated by pagination. The results are added to the output
+                # list of rows after each request.
+                for next_request, param_value_map in prepared_requests:
+                    while next_request:
+                        processed_rows, next_request = self.execute_prepared_request(  # type: ignore
+                            next_request,
+                            privacy_request.get_cached_identity_data(),
+                            read_request,
+                        )
+                        rows.extend(
+                            self._apply_output_template(
+                                [param_value_map],
+                                read_request.output,
+                                processed_rows,
+                            )
+                        )
+
+            # This allows us to build an output object even if we didn't generate and execute
+            # any HTTP requests. This is useful if we just want to select specific input_data
+            # values to provide as row data to the mask_data function
+            if not read_request.path and read_request.output:
```

Comment:
nit: this may read a bit simpler and i think is equivalent? 
```suggestion
            elif read_request.output:
```